cmd/state-svc/main.go

```diff
@@ -167,20 +159,25 @@ func run(cfg *config.Instance) (rerr error) {
 				if err := auth.Sync(); err != nil {
 					logging.Warning("Could not sync authenticated state: %s", err.Error())
 				}
-				return runForeground(cfg, an, checker)
+				return runForeground(cfg, an, auth)
 			},
 		),
 	)
 
 	return cmd.Execute(args[1:])
 }
 
-func runForeground(cfg *config.Instance, an *anaSvc.Client, deprecation *deprecation.Checker) error {
+func runForeground(cfg *config.Instance, an *anaSync.Client, auth *authentication.Auth) error {
 	logging.Debug("Running in Foreground")
 
-	// create a global context for the service: When cancelled we issue a shutdown here, and wait for it to finish
-	ctx, shutdown := context.WithCancel(context.Background())
-	p := NewService(cfg, an, deprecation, shutdown)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	p := NewService(ctx, cfg, an, auth)
+
+	if err := p.Start(); err != nil {
+		return errs.Wrap(err, "Could not start service")
+	}
```

Comment:
Trying to understand why this is being called now but wasn't before?

Reply:
Not sure how this showed up in the diff. It's already on master [here](https://github.com/ActiveState/cli/blob/master/cmd/state-svc/main.go#L175). The only thing I changed on the state-svc main file is to handle an error that we were ignoring.